force-app/main/default/objects/CellOrder__c/validationRules/Lock_fields_after_delivery_completed.validationRule-meta.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ValidationRule xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>Lock_fields_after_delivery_completed</fullName>
+    <active>true</active>
+    <errorConditionFormula>AND(
+    OR(
+        AND(ISPICKVAL(PRIORVALUE(State__c), &apos;Delivered&apos;), ISPICKVAL(State__c, &apos;Completed&apos;)),
+        AND(ISPICKVAL(PRIORVALUE(State__c), &apos;Completed&apos;), ISPICKVAL(State__c, &apos;Delivered&apos;)),
+        AND(ISPICKVAL(PRIORVALUE(State__c), &apos;Delivered&apos;), ISPICKVAL(State__c, &apos;Delivered&apos;)),
+        AND(ISPICKVAL(PRIORVALUE(State__c), &apos;Completed&apos;), ISPICKVAL(State__c, &apos;Completed&apos;))
```

Comment:
It is too complicated logic. We can check only prior value equals to "Delivered" and "Completed".